ar-setup.sh

```diff
@@ -0,0 +1,165 @@
+#!/bin/bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+artifact_registry_location=us-central1
+project=$(gcloud config get-value core/project)
+readonly project_var_name_gcp_ar_repo=GCP_AR_REPO
+readonly project_var_name_gcp_ar_key=GCP_AR_KEY
+
+usage () {
+    echo "usage: ./ar-setup.sh --app-name <app_name> --gitlab-access-token <token> --app-config-repo <app-config-repo> [--project <gcp_project>] [--artifact-registry-location <location>]"
+	echo "  OR   ./ar-setup.sh --app-name=<app_name> --gitlab-access-token=<token> --app-config-repo=<app-config-repo> [--project=<gcp_project>] [--artifact-registry-location=<location>]"
+	echo "  --app-name (required): the application name."
+	echo "  --app-config-repo (required): the URL to the application project. This is needed so that we can add the AR repository and the Google service account key as project variables of the application project."
+	echo "  --gitlab-access-token (required): the GitLab personal access token for accessing the GitLab APIs. If you don't have one, see https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html for how to create personal access tokens with 'api' scope."
+	echo "  --project (optional): the GCP project ID. If not specified, using the output of \`gcloud config get-value core/project\`."
+	echo "  --artifact-registry-location (optional): the artifact registry location (default to us-central1)."
+}
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+		--app-name) shift; app_name="$1" ;;
+		--app-name=*) app_name="${1#*=}" ;;
+		--project) shift; project="$1" ;;
+		--project=*) project="${1#*=}" ;;
+		--artifact-registry-location) shift; artifact_registry_location="$1" ;;
+		--artifact-registry-location=*) artifact_registry_location="${1#*=}" ;;
+		--app-config-repo) shift; app_config_repo="$1" ;;
+		--app-config-repo=*) app_config_repo="${1#*=}" ;;
+		--gitlab-access-token) shift; gitlab_access_token="$1" ;;
+		--gitlab-access-token=*) gitlab_access_token="${1#*=}" ;;
+		*) echo "Unknown parameter passed: $1"; usage; exit 1 ;;
+    esac
+    shift
+done
+
+if [ -z "${app_name}" ] || [ -z "${app-config-repo}" ] || [ -z "${gitlab_access_token}" ]; then
+	usage
```

Comment:
nit (optional): bring up usage when `--help` or `-h` is passed

Reply:
Done.